### PR TITLE
Fixed custom widget validation (#12629)

### PR DESCRIPTION
**Pull request by lexkrstn**
| Q | A |
| --- | --- |
| Is bugfix? | yes |
| New feature? | no |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | #12629 |

### CODE REVIEW

File: framework/widgets/ActiveField.php [hunk 1]
```diff
@@ -697,6 +697,9 @@ public function widget($class, $config = [])
         $config['model'] = $this->model;
         $config['attribute'] = $this->attribute;
         $config['view'] = $this->form->getView();
+        if (in_array('yii\widgets\InputWidget', class_parents($class)) && isset($config['options'])) {
```
Comment (samdark) [inline framework/widgets/ActiveField.php]: Couldn't it be checked like `$class instanceof yii\widgets\InputWidget`?

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/widgets/ActiveField.php (hunk 1 above)
Comment (coderlex) [inline framework/widgets/ActiveField.php]: Type of `$class` variable is string. The expression will ever return false.

Comment (samdark) [inline framework/widgets/ActiveField.php]: Right.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

File: framework/widgets/ActiveField.php (hunk 1 above)
Comment (rob006) [inline framework/widgets/ActiveField.php]: This should be more efficient:
```php
if (isset($config['options']) && isset(class_parents($class)['yii\widgets\InputWidget'])) {
```

**Pull request merged**